docs/reference/elasticsearch/rest-apis/api-conventions.md

```diff
@@ -421,6 +421,40 @@ GET /_nodes/ra*:2
 GET /_nodes/ra*:2*
 ```
 
+### Index Component Selectors [api-component-selectors]
+
+Some APIs that accept a `<data-stream>`, `<index>`, or `<target>` request path parameter also support *selector syntax*.
+
+Some indices and data streams are made of multiple components besides their default data. These other components store additional data which can be queried separately like a regular index. Some APIs may operate on subsets of these index components which can be specified by using a selector suffix.
+
+There are currently two selector suffixes supported by {{es}} APIs:
+
+`::data`
+:   (Indices, data streams) This selector references the default component present on all indices and data streams which holds the regular data contained therein.
+
+`::failures`
+:   (Data streams only) This component references data stored in a data stream's [failure store](docs-content://manage-data/data-store/data-streams/failure-store.md). This component is only present on data streams.
```

Comment:
I don't think we can say it references "data", because that doesn't work for non-data APIs. For example `GET /myds::failures/_settings` returns the settings for the failure store indices, and not the data stored in those failure store indices.